src/http.ts

```diff
@@ -41,24 +43,26 @@ export interface ConnectionOptions {
 // The trick to typecast union type alias
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 function isSubsetErrorCode<T>(code: any, messages: { [c: number]: string }): code is T {
-  if (code in messages) {
-    return true;
-  }
-  return false;
+  return code in messages;
 }
 
 function generateError<E>(
   errorCode: number,
   messages: { [c: number]: string },
   apiName: string,
-  error?: string,
+  errorMessage?: string,
+  errors?: unknown,
 ): ResultError<E> {
+  if (isJsonApiErrors(errors)) {
+    return generateJsonApiError<E>(errors, errorCode, apiName);
+  }
+
   if (!isSubsetErrorCode<E>(errorCode, messages)) {
-    throw { errorCode, messages, apiName };
+    throw { statusCode: errorCode, statusText: errorMessage || 'unknown error occurred', apiName };
```

Comment:
`q`: Not sure how this is handled, we're throwing this but it's not clear how downstream callers deal with it, are we just going to the global error handler in the CLI for example?

Reply:
I don't think the previous implementation ever worked, when it is thrown the CLI just output `[Object object]` in the console as the thrown object was used as the message in the [default error handling](https://github.com/snyk/cli/blob/f88d296daaf3b09d236f34af5008542522454f66/src/lib/plugins/sast/index.ts#L123). This make's the error handle-able by the [code client error type](https://github.com/snyk/cli/blob/f88d296daaf3b09d236f34af5008542522454f66/src/lib/plugins/sast/index.ts#L136) in the CLI.